app/services/discovery_engine/sync/locking.rb

```diff
@@ -49,6 +49,18 @@ def with_locked_document(content_id, payload_version:, &critical_section)
       raise FailedToAcquireLockError, "Failed to acquire lock for document: #{content_id}"
     end
 
+    # Determines if a given payload version is newer than the latest synced version for a document.
+    def newer_than_remote?(content_id, payload_version:)
```

Comment:
Minor: Perhaps the name could be `payload_newer_than_remote?`, then you wouldn't need the comment.